x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_form.tsx

```diff
@@ -149,6 +149,10 @@ export const AlertForm = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
```

Comment:
Is this changes related to the current issue? Did you want to fix something by adding it? 
I suppose it can replace the default state from this [line](https://github.com/elastic/kibana/blob/821e050672fa65666dc20f488057405ee40149ad/x-pack/plugins/triggers_actions_ui/public/application/sections/alert_form/alert_form.tsx#L109)

Reply:
This change was to handle a pre-selected alert being passed into the alert form (as in the case of uptime alerts). Previously, there was a delay in rendering the alert form because the health check loading indicator rendered first (however briefly). That allows the pre-selected alert to render when the alert form was first rendered. With the change to show the alert form immediately in the add alert flyout, this useEffect is needed to re-render the form to show the selected alert type.